app/src/main/java/com/github/se/polyfit/ui/screen/MapScreen.kt

```diff
@@ -114,39 +139,102 @@ fun MapScreen(
         GoogleMap(
             cameraPositionState = cameraPositionState,
-            properties = MapProperties(isMyLocationEnabled = true)) {
-              Circle(
-                  center = LatLng(currentLocation.value.latitude, currentLocation.value.longitude),
-                  radius =
-                      (sliderPosition).toDouble() + 10, // Make the shadow circle slightly larger
-                  strokeColor =
-                      Color.Black.copy(alpha = 0.2f) // Use a semi-transparent color for the shadow
-                  ,
-                  visible = isCircleShown)
-              Circle(
-                  center = LatLng(currentLocation.value.latitude, currentLocation.value.longitude),
-                  radius = (sliderPosition).toDouble(),
+            properties = MapProperties(isMyLocationEnabled = true),
+            onMapClick = { selectedPost = null }) {
+              val latLng = LatLng(currentLocation.value.latitude, currentLocation.value.longitude)
+              val doubleSliderPosition = sliderPositionInMeters.toDouble()
+              CircleOnMap(
+                  center = latLng,
+                  radius = doubleSliderPosition,
                   strokeColor = PrimaryPurple,
                   visible = isCircleShown)
-              listMarker.forEach { markerState ->
-                Marker(
-                    state = markerState,
-                    title = "title",
-                    contentDescription = "description",
-                    onClick = { goToMarker(goToPost, markerState) })
-              }
+              CircleOnMap(
+                  center = latLng,
+                  radius = doubleSliderPosition + 10,
+                  strokeColor = Color.Black.copy(alpha = 0.2f),
+                  visible = isCircleShown)
+
+              Marker(listMarker) { post -> selectedPost = post }
             }
-        Row(modifier = Modifier.padding(end = 64.dp)) {
-          Slider(
-              value = sliderPosition,
-              onValueChange = { sliderPosition = it },
-              valueRange = MIN_SIZE_RAYON..MAX_SIZE_RAYON,
-              modifier = Modifier.align(Alignment.CenterVertically).widthIn(10.dp, 200.dp))
+
+        SliderView(
+            sliderPosition = sliderPositionInMeters,
+            onSliderChanged = ::onSliderChanged,
+            updateSliderPosition = { sliderPositionInMeters = it },
+            isCircleShown = { isCircleShown = it })
+
+        selectedPost?.let { post ->
+          Box(modifier = Modifier.fillMaxWidth().padding(16.dp).align(Alignment.BottomCenter)) {
+            PostCard(post)
+          }
         }
       }
 }
 
-fun goToMarker(goToPost: () -> Unit, marker: MarkerState): Boolean {
-  goToPost()
-  return true
+@Composable
+@GoogleMapComposable
+fun CircleOnMap(center: LatLng, radius: Double, strokeColor: Color, visible: Boolean) {
+
+  Circle(center = center, radius = radius, strokeColor = strokeColor, visible = visible)
+}
```

Comment:
This is a great initiative!

could add optional parameters for more versatility
```
@Composable
@GoogleMapComposable
fun CircleOnMap(center: LatLng, radius: Double, strokeColor: Color, visible: Boolean, alpha: Float = 1f) {
    Circle(center = center, radius = radius, strokeColor = strokeColor.copy(alpha = alpha), visible = visible)
}
```
But since they aren't used currently it doesn't really change much

Reply:
?